tests/migration_test.go

```diff
@@ -1138,6 +1173,76 @@ var _ = Describe("[rfe_id:393][crit:high[vendor:cnv-qe@redhat.com][level:system]
 
 			})
 		})
+
+		Context("without CPU properly set", func() {
+
+			prepareMigration := func(cpuModel string) (*v1.VirtualMachineInstance, *v1.VirtualMachineInstanceMigration) {
+				vmi := tests.NewRandomVMIWithEphemeralDisk(tests.ContainerDiskFor(tests.ContainerDiskCirros))
+				vmi.Spec.Domain.CPU = &v1.CPU{
+					Model: cpuModel,
+				}
+				tests.AddUserData(vmi, "cloud-init", "#!/bin/bash\necho 'hello'\n")
+
+				By("Starting the VirtualMachineInstance")
+				vmi = runVMIAndExpectLaunch(vmi, 240)
+
+				By("Checking that the VirtualMachineInstance console has expected output")
+				expecter, err := tests.LoggedInCirrosExpecter(vmi)
+				Expect(err).ToNot(HaveOccurred())
+				expecter.Close()
+
+				// execute a migration, wait for finalized state
+				migration := tests.NewRandomMigration(vmi.Name, vmi.Namespace)
+
+				return vmi, migration
+			}
+
+			disposeVMI := func(vmi *v1.VirtualMachineInstance) {
+				// delete VMI
+				By("Deleting the VMI")
+				Expect(virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})).To(Succeed())
+
+				By("Waiting for VMI to disappear")
+				tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 240)
+			}
+
+			table.DescribeTable("should block migration with feature-gate CPUMigrationGate enabled", func(cpuModel string) {
+				tests.EnableFeatureGate("MigrationCPU")
+
+				vmi, migration := prepareMigration(cpuModel)
+
+				By("Starting a Migration")
+				migration, err = virtClient.VirtualMachineInstanceMigration(migration.Namespace).Create(migration)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("CPUNotMigratable"))
+
+				disposeVMI(vmi)
```

Comment:
optional nit: put this into an [`AfterEach`](http://onsi.github.io/ginkgo/#extracting-common-setup-beforeeach), and then remove this line and 1238 altogether.

Reply:
We can do this in further cleaning. The tests deserve more cleanup. I will create issues covering this topic.